src/viur/datastore/cache.py

```diff
@@ -0,0 +1,57 @@
+import sys
+from typing import Any, Dict, List, Union
+
+from viur.datastore.config import conf
+from viur.datastore.types import Entity, Key
+
+memcache_max_batch_size = 30
+memcache_namespace = "viur-datastore"
+memcache_timeout = 60 * 60
+memcache_max_size = 1_000_000
+
+
+def get(keys: Union[str, Key, List[str], List[Key]]):
+	if not isinstance(keys, list):
+		keys = [keys]
+	keys = [str(key) for key in keys]  # Enforce that all keys are strings
+	res = {}
+
+	while keys:
+		res |= conf["memcache_client"].get_multi(keys[:memcache_max_batch_size], namespace=memcache_namespace)
+		keys = keys[memcache_max_batch_size:]
+	return res
+
+
+def set(cache_data: Union[Dict, List[Entity]]):
+	if isinstance(cache_data, dict):
+		# Add only values to cache <= memcache_max_size (1.000.000)
+		cache_data = {str(key): value for key, value in cache_data.items() if get_size(value) <= memcache_max_size}
+	elif isinstance(cache_data, list):
+		# Add only values to cache <= memcache_max_size (1.000.000)
+		cache_data = {str(item.key): item for item in cache_data if get_size(item) <= memcache_max_size}
```

Comment:
less duplicate code and adds a type check. (not tested :grin:)
```suggestion
    if isinstance(cache_data, list):
        cache_data = {item.key: item for item in cache_data}
    elif not isinstance(cache_data, dict):
        raise TypeError(f"Invalid type {type(cache_data)}. Expected a list or dict.")
    # Add only values to cache <= memcache_max_size (1.000.000)
    cache_data = {str(key): value for key, value in cache_data.items()
                  if get_size(value) <= memcache_max_size}
```

Reply:
Mhh i think i must rework  this part